redash/query_runner/snowflake.py

```diff
@@ -96,16 +112,43 @@ def _get_connection(self):
             else:
                 host = "{}.snowflakecomputing.com".format(account)
 
-        connection = snowflake.connector.connect(
-            user=self.configuration["user"],
-            password=self.configuration["password"],
-            account=account,
-            region=region,
-            host=host,
-            application="Redash/{} (Snowflake)".format(__version__.split("-")[0]),
-        )
+        connection_parameters = {
+            "user": self.configuration["user"],
+            "password": self.configuration["password"],
+            "account": account,
+            "region": region,
+            "host": host,
+            "application": "Redash/{} (Snowflake)".format(__version__.split("-")[0]),
+            "session_parameters": {
+                "QUERY_TAG": "redash",
+            }
+        }
+
+        private_key_content = os.getenv("SNOWFLAKE_PRIVATE_KEY_CONTENT")
+        private_key_password = os.getenv("SNOWFLAKE_PRIVATE_KEY_PASSWORD")
+
+        if private_key_content and private_key_password:
+
+            p_key = serialization.load_pem_private_key(
+                private_key_content.encode(),
+                password=private_key_password.encode(),
+                backend=default_backend()
+            )
+
+            pkb = p_key.private_bytes(
+                encoding=serialization.Encoding.DER,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+
+            connection_parameters['private_key'] = pkb
+
+            # password is not needed
+            connection_parameters.pop('password')
+            logging.debug("Connectiong with the private key from SNOWFLAKE_PRIVATE_KEY_CONTENT and SNOWFLAKE_PRIVATE_KEY_PASSWORD env vars")
+
```

Comment:
Is this just saying that it uses the connection parameters if the os variables aren't found?

Reply:
yeah. just in case